lib/saxy/encoder.ex

```diff
@@ -34,7 +64,7 @@ defmodule Saxy.Encoder do
     [?\s, 'standalone', ?=, ?", "yes", ?"]
   end
 
-  defp element({tag_name, attributes, :empty}) do
+  defp element({tag_name, attributes, []}) do
```

Comment:
I agree that using `:empty` here might not be the best. How about `nil` instead of `[]`? Self-closing tags aren't semantically the same as tags with empty content IMO.

Reply:
I prefer to keep the same kind of data that way I have to do less processing when processing children.

Reply:
That's fair I think. We should still make it possible if anyone wants to generate tags with empty content though 🤔? It's supported currently with `Saxy.XML.empty_element/2` interface.

Would it make sense to use `{tag_name, attributes}` for self-closing and `{tag_name, attributes, []}` for empty content tag?

Reply:
actually, for XML `<foo></foo>` should be completely equivalent to `<foo/>` AFAIK

Reply:
Yeah seems like that’s also mentioned in the XML 1.0 specs.